protocol/pubsub/v2/message.go

```diff
@@ -120,14 +121,13 @@ func (m *Message) GetExtension(name string) interface{} {
 }
 
 // Finish marks the message to be forgotten.
-// If err is nil, the underlying Pubsub message will be acked;
-// otherwise nacked and return the error.
+// Regarding protocol.IsAck, the Pubsub message will be acked or not.
```

Comment:
I suggest changing the description to:

```
Finish marks the message to be forgotten and returns the provided error without modification. If err is nil or of type protocol.ResultACK the PubSub message will be acknowledged, otherwise nack-ed.
```

Reply:
Thinking more about this here, I think we should always return nil and document this here because otherwise the caller could think that finish returned an error which is not the case. WDYT?

Reply:
replied here: https://github.com/cloudevents/sdk-go/pull/1064#discussion_r1665418583

Reply:
@embano1
I've updated the pubsub doc according to your comment.

About your suggestion for the Finish function error processing:

Thinking more about this here, I think we should always return nil and document this here because otherwise the caller could think that finish returned an error which is not the case. WDYT?

I agree this is "weird" currently but note that the error returned is actually used to perform some logging...
https://github.com/cloudevents/sdk-go/blob/main/v2/client/client.go#L170

I suggest we open another Issue to discuss these topics:

do not forward the error in Finish
return error in Finish only for real acknowledgement error
add an additional "status" result to share all processing info with ceClient in shared layers (logs, observability, ...)